Backend/src/main/java/com/example/Backend/controller/UserController.java

```diff
@@ -0,0 +1,136 @@
+            @ApiResponse(responseCode = "200", description = "User Fetched Successfully")
+    })
+    @GetMapping("/user/{name}")
+    public ResponseEntity<UserLogin> displayUser(@PathVariable(value = "name") String name) {
+        if(name.isEmpty())
+        {
+            log.info("Invalid Request");
+            return new ResponseEntity<>(UserLogin.builder().username("Empty Username").build(),HttpStatus.BAD_REQUEST);
+        }
+        else {
+            UserLogin userLogin = this.loginService.getUserDetails(name);
+            if (userLogin.getUsername().equals("Not_Found")) {
+                log.error("User Doesn't exists");
+                return new ResponseEntity<>(userLogin, HttpStatus.NOT_FOUND);
+            } else {
+                log.info("User Fetched Successfully: {}", userLogin.getUsername());
+                return new ResponseEntity<>(userLogin, HttpStatus.OK);
+            }
+        }
+    }
+
+    @Operation(summary = "Login user", description = "Authenticate user with provided credentials")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "202", description = "User Logged in Successfully!"),
+            @ApiResponse(responseCode = "404", description = "User Doesn't Exist!"),
+            @ApiResponse(responseCode = "401", description = "Wrong Password!")
+
+    })
+    @PostMapping("/login")
+    public ResponseEntity<String> loginUser(@Valid @RequestBody UserLogin userLogin, BindingResult bindingResult) {
+        StringBuilder errorMessage = new StringBuilder();
+        try {
+            String loginServiceResponse = this.loginService.login(userLogin);
+            if (loginServiceResponse.equals("Found")) {
+                log.info("User Login: {}", userLogin.getUsername());
+                return new ResponseEntity<>("User Logged in Successfully!", HttpStatus.ACCEPTED);
+            }
+            else {
+                if (bindingResult.hasErrors()) {
+                    List<FieldError> errors = bindingResult.getFieldErrors();
+                    errorMessage = new StringBuilder("Validation errors: ");
+                    for (FieldError error : errors) {
+                        errorMessage.append(error.getDefaultMessage()).append("\n");
+                    }
+                    throw new MethodArgumentNotValidException(null, bindingResult);
+                }
+                else if(loginServiceResponse.equals("Wrong Password"))
+                {
+                    log.info("Wrong Password for user: {}", userLogin.getUsername());
+                    return new ResponseEntity<>("Wrong Password!", HttpStatus.UNAUTHORIZED);
+                }
+                else if(loginServiceResponse.equals("Not Found"))
+                {
+                    log.info("User Doesn't Exists!");
+                    return new ResponseEntity<>("User Doesn't Exist!", HttpStatus.NOT_FOUND);
+                }
+            }
+        } catch (MethodArgumentNotValidException e) {
```

Comment:
Where and when is this exception thrown?

Reply:
line 82: MethodArgumentNotValidException is thrown is it receive message{from login model} in case of Null input.